pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -50,6 +51,23 @@ type Controller struct {
 	model.NetworkGatewaysHandler
 }
 
+func (c *Controller) PodInformation(podsUpdated map[model.ConfigKey]struct{}) ([]model.WorkloadInfo, []string) {
+	i := []model.WorkloadInfo{}
+	removed := sets.New()
+	for _, p := range c.registries {
+		wis, r := p.PodInformation(podsUpdated)
```

Comment:
Ok for now - but I think the reverse map from IP to workload info ( and from VIP to service info ) should be core in the model, and updated when the pod is changing ( without iterating trough all registries/pods/etc). 

Also I would add a TODO to include the network name somewhere.

Reply:
I initially thought rust was using on-demand already. some of my comments were based on that. If we send all endpoints - this is ok.